app/registries/foreman/setting_manager.rb

```diff
@@ -71,6 +79,37 @@ def setting(name, default:, description:, type:, full_name: nil, value: nil, col
           encrypted: encrypted,
           options: options,
         }
+        validates(name, validate) if validate
+        storage[name.to_s]
+      end
+
+      def validates(name, validations)
+        if validations.is_a?(Proc)
+          validates_with name, ValueLambdaValidator, proc: validations
+          return
+        elsif validations.is_a?(Regexp)
+          validations = { format: { with: validations } }
+        elsif validations.is_a?(Symbol)
+          validations = { validations => true }
+        end
+        _wrap_validation_if(name, validations)
+        Setting.validates(:value, validations)
+      end
+
+      def validates_with(name, *args, &block)
+        options = args.extract_options!
+        _wrap_validation_if(name, options)
+        options[:attributes] = [:value]
+        args << options
+        Setting.validates_with(*args, &block)
+      end
+
+      def _wrap_validation_if(setting_name, options)
+        options[:if] = [
+          ->(setting) { setting.name == setting_name.to_s },
+          *options[:if],
+        ]
+        options[:allow_blank] = storage[setting_name.to_s].dig(:options, :allow_blank) || false
```

Comment:
This line doesn't seem related to this function

Reply:
I've refactored it